pkg/apis/pipeline/v1/task_validation_test.go

```diff
@@ -288,6 +288,33 @@ func TestTaskSpecValidate(t *testing.T) {
 				WorkingDir: "/foo/bar/src/",
 			}},
 		},
+	}, {
+		name: "valid results path variable in script",
+		fields: fields{
+			Steps: []v1.Step{{
+				Name:  "step-name",
+				Image: "my-image",
+				Script: `
+				#!/usr/bin/env bash
+				date | tee $(results.a-result.path)`,
+			}},
+			Results: []v1.TaskResult{
+				{
+					Name: "a-result",
+				},
+			},
+		},
+	}, {
+		name: "valid results in step image",
```

Comment:
should there also be a test case for an invalid result in an image?

Reply:
NIT: this test case is not necessary and might create confusion in future that a step `image` does accept `results`. This is anyways invalid, since `a-result` is not declared as a result.